hosted/gke/p1/p1_provisioning_test.go

```diff
@@ -306,4 +313,71 @@ var _ = Describe("P1Provisioning", func() {
 			updateClusterInUpdatingState(cluster, ctx.RancherAdminClient)
 		})
 	})
+
+	When("a private cluster is created", func() {
+
+		var err error
+		var updateFunc func(clusterConfig *gke.ClusterConfig)
+		network := pointer.String("hosted-providers-ci-private")
+
+		BeforeEach(func() {
+			updateFunc = func(clusterConfig *gke.ClusterConfig) {
+				clusterConfig.Network = network
+				clusterConfig.Subnetwork = network
+				clusterConfig.PrivateClusterConfig.EnablePrivateNodes = true
+				clusterConfig.PrivateClusterConfig.MasterIpv4CidrBlock = fmt.Sprintf("172.16.%d.0/28", rand.Intn(10))
+
+				currentSpec := CurrentSpecReport().FullText()
+				if strings.Contains(currentSpec, "MasterAuthorizedNetworks") {
+					_, authorizedIP, err := net.ParseCIDR(strings.TrimSuffix(helpers.RancherHostname, ".sslip.io") + "/32")
```

Comment:
This will not work if the hostname does not follow convention as our GH runners. For e.g., I use EC2 public DNS as my rancher hostname `ec2-12-345-678-910.ap-south-1.compute.amazonaws.com` will not work and error out.

Reply:
Added a generic function for this.